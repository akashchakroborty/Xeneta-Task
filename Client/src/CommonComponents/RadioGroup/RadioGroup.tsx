import React from 'react';
import { Radio, RadioChangeEvent } from 'antd';

export interface RadioGroupProps {
  onChange: (e: RadioChangeEvent) => void;
  selectedValue: string;
  data: Map<string, string>;
}

const RadioGroupComponent: React.FC<RadioGroupProps> = ({ onChange, selectedValue, data }) => {
  return (
    <Radio.Group onChange={onChange} value={selectedValue}>
      {[...data.keys()].map((value) => (
        <Radio value={data.get(value)!} key={data.get(value)}>
          {value}
        </Radio>
      ))}
    </Radio.Group>
  );
};

export default RadioGroupComponent;
