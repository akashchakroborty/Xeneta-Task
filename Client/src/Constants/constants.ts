export const SELECT_ORIGIN = 'Select Origin';
export const SELECT_DESTINATION = 'Select Destination';
export const EMPTY_STRING = '';
export const DEFAULT_PORTS = [];
export const DEFAULT_RATES = [];
export const DEFAULT_LOADING = false;
