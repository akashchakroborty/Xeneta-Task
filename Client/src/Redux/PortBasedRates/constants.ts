export const SET_PORTS_SUCCESS = 'SET_PORTS_SUCCESS';
export const SET_PORTS_FAILURE = 'SET_PORTS_FAILURE';
export const GET_PORTS_LOADING = 'GET_PORTS_LOADING';
export const UPDATE_ORIGIN = 'UPDATE_ORIGIN';
export const UPDATE_DESTINATION = 'UPDATE_DESTINATION';
