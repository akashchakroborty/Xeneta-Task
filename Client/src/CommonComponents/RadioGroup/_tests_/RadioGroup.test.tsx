import React from 'react';
import { shallow } from 'enzyme';
import RadioGroup, { RadioGroupProps } from '../RadioGroup';
import { DEFAULT_RATE_TYPE, RATE_TYPE_MAPPER } from '../../../Constants/constants';

describe('should test the RadioGroup Component', () => {
  const mockProps: RadioGroupProps = {
    onChange: jest.fn(),
    selectedValue: DEFAULT_RATE_TYPE,
    data: RATE_TYPE_MAPPER,
  };
  test('should render the Component', () => {
    const component = shallow(<RadioGroup {...mockProps} />);
    expect(component).toMatchSnapshot();
  });
});
