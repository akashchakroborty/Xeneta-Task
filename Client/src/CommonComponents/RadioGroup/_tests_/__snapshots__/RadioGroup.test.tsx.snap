// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test the RadioGroup Component should render the Component 1`] = `
<Memo(ForwardRef)
  onChange={[MockFunction]}
  value="mean"
>
  <Radio
    key="mean"
    type="radio"
    value="mean"
  >
    Market Average
  </Radio>
  <Radio
    key="low"
    type="radio"
    value="low"
  >
    Market Low
  </Radio>
  <Radio
    key="high"
    type="radio"
    value="high"
  >
    Market High
  </Radio>
</Memo(ForwardRef)>
`;
